src/canvas/PatternPath.tsx

```diff
@@ -94,6 +94,56 @@ export class PatternPath implements IPatternPath {
         this._path2D.quadraticCurveTo(fittedCurve.control.getX(), fittedCurve.control.getY(), fittedCurve.end.getX(), fittedCurve.end.getY());        
     }
 
+    snapEndpoints = (paths: PatternPath[]): void => {
+        const myFirstPoint = this._points[0];
+        const myLastPoint = this._points[this._points.length - 1];
+        // Radius to check within to see if we should snap to point.
+        const radius = 10;
+        let updatedFirstPoint = false;
+        let updatedLastPoint = false;
+
+        paths.forEach(path => {
+            if (path === this) {
+                return;
+            }
+            const points = path.getPoints();
+
+            const otherFirstPoint = points[0];
+            const otherLastPoint = points[points.length - 1];
+
+            if(!updatedFirstPoint && myFirstPoint.isWithinRadius(otherFirstPoint, radius)) {
+                this._points[0] = otherFirstPoint;
+                updatedFirstPoint = true;
+            }
+
+            if(!updatedFirstPoint && myFirstPoint.isWithinRadius(otherLastPoint, radius)) {
+                this._points[0] = otherLastPoint;
+                updatedFirstPoint = true;
+            }
+
+            if(!updatedLastPoint && myLastPoint.isWithinRadius(otherFirstPoint, radius)) {
+                this._points[this._points.length] = otherFirstPoint;
+                updatedLastPoint = true;
+            }
+
+            if(!updatedLastPoint && myLastPoint.isWithinRadius(otherLastPoint, radius)) {
+                this._points[this._points.length] = otherLastPoint;
+                updatedLastPoint = true;
+            }
+
+        });
+
+        if (updatedFirstPoint || updatedLastPoint) {
+            if (this._toolType === ToolType.StraightLine) {
+                this._updatePath2DStraightLine();
+            }
+
+            if (this._toolType === ToolType.Freeline) {
+                this._updatePath2DWithQuadraticCurve();
+            }
```

Comment:
Here's pseudocode for something that uses fewer if checks : 
```
updated = false
endpoints = []
paths.forEach(path => 
    endpoints.add(points[0])
    endpoints.add(points[last])
)
endpoints.foreach(endpoint => 
    if (endpoint.iswithinradius(myfirstpoint))
        this.points[0] = endpoint
        updated = true
        break
endpoints.foreach(endpoint => 
    if (endpoint.iswithinradius(mylastpoint))
        this.points[last] = endpoint
        updated = true
        break
if (updated)
    if (this._toolType === ToolType.StraightLine) {
          this._updatePath2DStraightLine();
     }

     if (this._toolType === ToolType.Freeline) {
         this._updatePath2DWithQuadraticCurve();
     }
```

Reply:
I actually did originally have multiple for loops, but removed them. In the end, this is doing the same number of condition checks, they just aren't each written out. I think I will keep it the way it is for readability.